resources/testdriver.js

```diff
@@ -0,0 +1,101 @@
+(function() {
+    "use strict";
+
+    function getInViewCenterPoint(rect) {
+        var left = Math.max(0, rect.left);
+        var right = Math.min(window.innerWidth, rect.right);
+        var top = Math.max(0, rect.top);
+        var bottom = Math.min(window.innerHeight, rect.bottom);
+
+        var x = 0.5 * (left + right);
+        var y = 0.5 * (top + bottom);
```

Comment:
Perhaps you wanted to use (right-left) and (top-bottom) here instead so that later you add the top and left of the element to them?

Reply:
This should match https://w3c.github.io/webdriver/webdriver-spec.html#dfn-center-point